src/App.php

```diff
@@ -161,6 +161,11 @@ function () use ( $slim ) {
 					$page->setDao( $slim->plagiabotDao );
 					$page->setEnwikiDao( $slim->enwikiDao );
 					$page->setWikiprojectDao( $slim->wikiprojectDao );
+
+					// determine if we are on the staging environment, so we can show a banner in the view
+					$rootUri = $slim->request->getRootUri();
+					$slim->view->set( 'staging', preg_match( '/\/plagiabot/', $rootUri ) );
```

Comment:
Not that it makes much difference since it's just a URL, but strpos() is better to use in this case than preg_match() as it's more efficient (it doesn't use the regex engine).